esti/s3_gateway_test.go

```diff
@@ -289,6 +289,134 @@ func TestS3IfNoneMatch(t *testing.T) {
+	if blockStoreType != "s3" {
+		require.Contains(t, err.Error(), "NotImplemented")
+		return
+	}
+	require.NoError(t, err, "error listing multiparts")
+	keys := extractUploadKeys(output)
+	require.Contains(t, keys, obj1)
+
+	// create second mpu check both appear
+	_, err = s3Client.CreateMultipartUpload(ctx, input2)
+	require.NoError(t, err, "failed to create multipart upload")
+	output, err = s3Client.ListMultipartUploads(ctx, &s3.ListMultipartUploadsInput{Bucket: resp1.Bucket})
+	keys = extractUploadKeys(output)
+	require.Contains(t, keys, obj1)
+	require.Contains(t, keys, obj2)
+
+	// testing maxuploads - only first upload should return
+	maxUploads := aws.Int32(1)
+	output, err = s3Client.ListMultipartUploads(ctx, &s3.ListMultipartUploadsInput{Bucket: resp1.Bucket, MaxUploads: maxUploads})
+	require.NoError(t, err, "failed to list multipart uploads")
+	keys = extractUploadKeys(output)
+	require.Contains(t, keys, obj1)
+	require.NotContains(t, keys, obj2)
+
+	// testing key marker and upload id marker for pagination. only records after marker should return
+	keyMarker := output.NextKeyMarker
+	uploadIDMarker := output.NextUploadIdMarker
+	output, err = s3Client.ListMultipartUploads(ctx, &s3.ListMultipartUploadsInput{Bucket: resp1.Bucket, MaxUploads: maxUploads, KeyMarker: keyMarker, UploadIdMarker: uploadIDMarker})
+	require.NoError(t, err, "failed to list multipart uploads")
+	keys = extractUploadKeys(output)
+	require.NotContains(t, keys, obj1)
+	require.Contains(t, keys, obj2)
+
+	// finish first mpu check only second appear
+	_, err = s3Client.CompleteMultipartUpload(ctx, completeInput1)
+	require.NoError(t, err, "failed to complete multipart upload")
+	output, err = s3Client.ListMultipartUploads(ctx, &s3.ListMultipartUploadsInput{Bucket: resp1.Bucket})
+	require.NoError(t, err, "error listing multiparts")
+	keys = extractUploadKeys(output)
+	require.NotContains(t, keys, obj1)
+	require.Contains(t, keys, obj2)
+
+}
+func TestListMultipartUploadsUnsupported(t *testing.T) {
+	blockStoreType := viper.GetString(ViperBlockstoreType)
+	if blockStoreType != "s3" {
+		return
+	}
+	ctx, _, repo := setupTest(t)
+	defer tearDownTest(repo)
+	s3Endpoint := viper.GetString("s3_endpoint")
+	s3Client := createS3Client(s3Endpoint, t)
+	Bucket := aws.String(repo)
+
+	delimiter := aws.String("/")
+	prefix := aws.String("prefix")
+	encodingType := types.EncodingTypeUrl
+
+	_, err := s3Client.ListMultipartUploads(ctx, &s3.ListMultipartUploadsInput{Bucket: Bucket, Delimiter: delimiter})
+	require.Error(t, err)
+	require.Contains(t, err.Error(), "NotImplemented")
+
+	_, err = s3Client.ListMultipartUploads(ctx, &s3.ListMultipartUploadsInput{Bucket: Bucket, Prefix: prefix})
+	require.Error(t, err)
+	require.Contains(t, err.Error(), "NotImplemented")
+
+	_, err = s3Client.ListMultipartUploads(ctx, &s3.ListMultipartUploadsInput{Bucket: Bucket, EncodingType: encodingType})
+	require.Error(t, err)
+	require.Contains(t, err.Error(), "NotImplemented")
```

Comment:
1. As far as I remember we don't capture unsupported states. Wonder if need this extra test as we don't do it for all the adapters.
2. Can use t.Run to capture each state